app/src/main/java/com/example/drawingactivity/DialogUtils.kt

```diff
@@ -115,31 +140,50 @@ object DialogUtils {
         sheet.apply {
             setOnDismissListener {
                 dismiss()
-                onDialogComplete?.invoke()
+                onSheetComplete?.invoke()
             }
             setContentView(view)
             show()
         }
         val behavior = BottomSheetBehavior.from(view.parent as View)
         behavior.state = BottomSheetBehavior.STATE_EXPANDED
         behavior.isDraggable = false
-        return { index, newName, operation ->
+        return { model, newModel, operation ->
             when (operation) {
+                DialogOperation.ADD -> {
+                    if (newModel != null) {
+                        data.add(newModel)
+                        recyclerView.adapter?.notifyItemInserted(data.size)
+                    }
+                }
                 DialogOperation.EDIT -> {
-                    data[index] = LayerViewModel(layerName = newName)
-                    recyclerView.adapter?.notifyItemChanged(index, newName)
+                    model?.let {
+                        val index = data.indexOf(model)
+                        if (newModel != null) {
+                            data[index] = newModel
+                            recyclerView.adapter?.notifyItemChanged(index, newModel)
+                        }
+                    }
                 }
                 DialogOperation.DELETE -> {
-                    data.removeAt(index)
-                    recyclerView.adapter?.notifyItemChanged(index)
+                    model?.let {
+                        val index = data.indexOf(model)
```

Comment:
nit: Can combine

```
data.indexOf(model).let { index -> ...
```

Reply:
Doesn't really get you anything. Just pointing it out

Reply:
oh damn that's pretty slick, I'll use that